### PR TITLE
Skip stale price update from keystone instead of reverting

### PR DESCRIPTION
**Pull request by 0xsuryansh**
Keystone report processing optimization, onReport should never revert deliberately

### CODE REVIEW

File: contracts/src/v0.8/ccip/FeeQuoter.sol [hunk 1]
```diff
@@ -520,7 +520,7 @@ contract FeeQuoter is AuthorizedCallers, IFeeQuoter, ITypeAndVersion, IReceiver,
 
       //if stale update then revert
       if (feeds[i].timestamp < s_usdPerToken[feeds[i].token].timestamp) {
-        revert StaleKeystoneUpdate(feeds[i].token, feeds[i].timestamp, s_usdPerToken[feeds[i].token].timestamp);
```
Comment (RyanRHall) [inline contracts/src/v0.8/ccip/FeeQuoter.sol:523]: can we remove this error entirely from the file now?

File: contracts/src/v0.8/ccip/test/feeQuoter/FeeQuoter.t.sol [hunk 2]
```diff
@@ -2359,13 +2359,15 @@ contract FeeQuoter_onReport is FeeQuoter_KeystoneSetup {
 
     //create a stale report
     report[0] =
-      FeeQuoter.ReceivedCCIPFeedReport({token: onReportTestToken1, price: 4e18, timestamp: uint32(block.timestamp - 1)});
+              FeeQuoter.ReceivedCCIPFeedReport({token: onReportTestToken1, price: 4e18, timestamp: uint32(block.timestamp - 1)});
+
+    //record logs to check no events were emitted
+    vm.recordLogs();
+
     //expecting a revert
```
Comment (RyanRHall) [inline contracts/src/v0.8/ccip/test/feeQuoter/FeeQuoter.t.sol]: this comment is no longer accurate, right?

**Review by RyanRHall — CHANGES_REQUESTED** [review]

File: contracts/src/v0.8/ccip/FeeQuoter.sol [hunk 3]
```diff
@@ -520,7 +519,7 @@ contract FeeQuoter is AuthorizedCallers, IFeeQuoter, ITypeAndVersion, IReceiver,
 
       //if stale update then revert
```
Comment (RensR) [inline contracts/src/v0.8/ccip/FeeQuoter.sol]: stale comment, please update with the reasoning why we skip stale prices

File: contracts/src/v0.8/ccip/test/feeQuoter/FeeQuoter.t.sol [hunk 4]
```diff
@@ -2339,7 +2339,7 @@ contract FeeQuoter_onReport is FeeQuoter_KeystoneSetup {
     s_feeQuoter.onReport(encodedPermissionsMetadata, abi.encode(report));
   }
 
-  function test_OnReport_StaleUpdate_Revert() public {
+  function test_OnReport_StaleUpdate_SkipPriceUpdate() public {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/feeQuoter/FeeQuoter.t.sol]: nit: success cases should always end with `success`

**Review by elatoskinas — DISMISSED** [review]
LGTM, after resolving comments from Rens

File: contracts/src/v0.8/ccip/FeeQuoter.sol (hunk 3 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/FeeQuoter.sol]: https://github.com/smartcontractkit/chainlink/pull/14981/commits/ded0b296ced82eb3f14a1e340677644dbf52cf47#diff-1859122b5ff5261692222f5ba921ee9108438593b0664e5aba8d573d866f27aaL520-R521

File: contracts/src/v0.8/ccip/test/feeQuoter/FeeQuoter.t.sol (hunk 4 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/test/feeQuoter/FeeQuoter.t.sol]: https://github.com/smartcontractkit/chainlink/pull/14981/commits/ded0b296ced82eb3f14a1e340677644dbf52cf47#diff-3df5401c7492ce02a335a3827289d64b15e207044c2e5dbc5337f18e7c30e4e7R2297

**Review by elatoskinas — APPROVED** [review]

**Review by RyanRHall — APPROVED** [review]

**Review by jhweintraub — APPROVED** [review]

**Pull request merged**